src/crimson/os/seastore/cache.h

```diff
@@ -288,20 +290,21 @@ class Cache : public ExtentTransViewRetriever {
     }
 
     ceph_assert(ret->get_type() == type);
+    t.add_to_read_set(ret);
+    touch_extent(*ret, &t_src, t.get_cache_hint());
     if (!ret->is_fully_loaded()) {
-      // ignore non-full extent
       SUBDEBUGT(seastore_cache,
         "{} {}~0x{:x} is present without fully loaded in cache -- {}",
         t, type, paddr, len, *ret);
-      return get_extent_if_cached_iertr::make_ready_future<CachedExtentRef>();
+      return trans_intr::make_interruptible(
+        do_read_extent_maybe_partial<CachedExtent>(
```

Comment:
I think the philosophy behind this change is that, since `get_extent_if_cached` is called during cleaning which always rewrite the extents got from here, we can directly load the data. Am I right?

Reply:
Yeah, and I think more generally (without assuming it is only called during cleaning) it should be fine under the protection of transaction conflicts of the present extent.